libbeat/processors/actions/changeFieldProcessor.go

```diff
@@ -0,0 +1,89 @@
+package actions
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/pkg/errors"
+)
+
+// fieldProccessorFunc defines how fields must be processed
+type fieldProccessorFunc func(event *beat.Event, field string) error
+
+type changeFieldProcessor struct {
+	Fields        []string
+	IgnoreMissing bool
+	FailOnError   bool
+	fieldProcess  string
+	changeFunc    fieldProccessorFunc
+}
+
+// NewChangeFieldProcessor is an umbrella method for processing events based on provided fields. Such as converting event keys to uppercase/lowercase
+func NewChangeFieldProcessor(c *conf.C, fieldProcess string, changeFunc fieldProccessorFunc) (processors.Processor, error) {
+	config := struct {
+		Fields        []string `config:"fields"`
+		IgnoreMissing bool     `config:"ignore_missing"`
+		FailOnError   bool     `config:"fail_on_error"`
+	}{
+		IgnoreMissing: false,
+		FailOnError:   true,
```

Comment:
Just FYI: the reason why these default values are like so is consistency with other processors, this processor should be identically configured to https://www.elastic.co/guide/en/beats/filebeat/current/rename-fields.html